src/etl/v2g/distance/distance.py

```diff
@@ -0,0 +1,69 @@
+"""The distance dataset contains gene distance to credible set variants provided that these are found within a certain region."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import pyspark.sql.functions as f
+
+if TYPE_CHECKING:
+    from pyspark.sql import DataFrame
+
+    from etl.common.ETLSession import ETLSession
+
+from etl.common.spark_helpers import normalise_column
+
+
+def main(
+    etl: ETLSession,
+    variant_index: DataFrame,
+    gene_index: DataFrame,
+    tss_distance_threshold: int,
+) -> DataFrame:
+    """Extracts variant to gene assignments for variants falling within a window of a gene's TSS."""
+    etl.logger.info("Generating distance related V2G data...")
+
+    return (
+        get_variant_distance_to_gene(gene_index, variant_index, tss_distance_threshold)
+        .select(
+            "*",
+            f.lit("distance").alias("datatypeId"),
+            f.lit("canonical_tss").alias("datasourceId"),
+        )
+        .transform(lambda df: score_distance(df, tss_distance_threshold))
+    )
+
+
+def score_distance(df: DataFrame, max_distance: int) -> DataFrame:
+    """Scores the distance between a variant and a gene's TSS.
+
+    The inverse of each distance is first calculated so that the closer the variant is to the TSS, the higher the score.
+
+    Args:
+        df (DataFrame): The V2G df based on distance.
+        max_distance (int): The maximum distance between a variant and a gene's TSS.
+
+    Returns:
+        DataFrame: The V2G df with the normalised score column.
+    """
+    inverse_expr = (max_distance - f.col("distance")).alias("tmp_score")
+    return (
+        df.select("*", inverse_expr)
+        .transform(lambda df: normalise_column(df, "tmp_score", "new_score"))
+        .drop("tmp_score")
+    )
+
+
+def get_variant_distance_to_gene(
+    gene_df: DataFrame,
+    variant_df: DataFrame,
+    distance_window: int,
+) -> DataFrame:
+    """Calculates the distance between a variant and the TSS of a gene."""
+    return (
+        variant_df.select("variantId", "chromosome", "position")
+        .join(f.broadcast(gene_df), on="chromosome", how="inner")
```

Comment:
I'm sure you have explored this, but I'm curious: if you join by the two dataframes by already specifying the condition (`f.abs(f.col("position") - f.col("tss")<= distance_window`) wouldn't be faster? The resulting dataframe would be way smaller and you would save on the filtering. But spark might optimize on the backend already.

Reply:
In main we have the same situation [here](https://github.com/opentargets/genetics_etl_python/blob/57b0adf9e68e10f4256826ea00f3b7c6de7dc915/src/etl/v2g/intervals/helpers.py#L42)

I don't know how smart Spark is with this things. But @DSuveges sounds like a better solution to me